radix-transactions/src/builder/manifest_namer.rs

```diff
@@ -978,25 +978,11 @@ pub trait ResolvableComponentAddress {
     fn resolve(self, registrar: &ManifestNameRegistrar) -> DynamicComponentAddress;
 }
 
-impl<'a> ResolvableComponentAddress for &'a str {
-    fn resolve(self, registrar: &ManifestNameRegistrar) -> DynamicComponentAddress {
-        registrar.name_lookup().named_address_id(self).into()
-    }
-}
-
-impl<'a> ResolvableComponentAddress for &'a String {
-    fn resolve(self, registrar: &ManifestNameRegistrar) -> DynamicComponentAddress {
-        registrar.name_lookup().named_address_id(self).into()
-    }
-}
-
-impl<'a> ResolvableComponentAddress for String {
-    fn resolve(self, registrar: &ManifestNameRegistrar) -> DynamicComponentAddress {
-        registrar.name_lookup().named_address_id(self).into()
-    }
-}
-
-impl<A: TryInto<DynamicComponentAddress, Error = E>, E: Debug> ResolvableComponentAddress for A {
+impl<A, E> ResolvableComponentAddress for A
```

Comment:
Out of interest, why this refactor?

We've been recently moving in the opposite direction, saving `where` bounds for only the complex bounds which require them.

Relevant context: https://github.com/radixdlt/radixdlt-scrypto/pull/1857 and [this slack thread](https://rdxworks.slack.com/archives/C071C6DB8TG/p1722037974593499) where we discussed:

> General guidelines:
> - Prefer impl T over a type parameter for input parameters (for new code, where it’s not a breaking change to remove a type parameter). A benefit is it reduces the number of type parameters where these are expected to just be inferred.
> - Prefer type parameter constraint over a where constraint where possible